### PR TITLE
Add cache-control, HSTS and HttpOnly headers

### PR DESCRIPTION
**Pull request by ambrozic**
According to pen test report we were advised to add this changes:
- Cache-Control headers
- X-XSS-Protection headers
- HttpOnly on csrftoken cookie

### CODE REVIEW

File: ui/middleware.py [hunk 1]
```diff
@@ -12,3 +12,11 @@ def process_request(self, request):
             if "runserver" not in sys.argv and "test" not in sys.argv:
                 return HttpResponseRedirect(urlunsplit(
                     ["https"] + list(urlsplit(request.get_raw_uri())[1:])))
+
+
+class CacheControlMiddleware(MiddlewareMixin):
+
+    def process_response(self, request, response):
```
Comment (elcct) [inline ui/middleware.py:19]: Should we have a test that this indeed is being added?

Comment (mcbhenwood) [reply ui/middleware.py:19]: Yep 😊

**Comment by ambrozic** [comment]
> For environments _not_ delivered over https (local / ci), will the HSTS headers still be sent on http-only responses? If so that will cause problems.

`Strict-Transport-Security` is only added to response headers if connection is over https
https://github.com/django/django/blob/master/django/middleware/security.py#L35-L36

This header is currently already there, i just added few more bits to it as per pen test report.

File: ui/middleware.py (hunk 1 above)
Comment (ambrozic) [reply ui/middleware.py:19]: test added. please, have another look.

**Pull request merged**